src/types.ts

```diff
@@ -1,140 +1,343 @@
+  createZDAOFromParams(param: CreateZDAOParams): Promise<void>;
+}
 
-  /**
-   * Get all the proposals added in the zDAO
-   * @param zNA zNA address
-   */
-  getProposalsByZDAOId(zNA: string, skip: number): Promise<Array<Proposal>>;
+export interface zDAO {
+  id: zDAOId; // Global zDAO identifier
+  zNA: zNA; // Linked zNA
+  title: string; // zDAO title, zNA by default
+  creator: string; // Creator wallet address
+  // avatar uri which starts with https schema
+  // The frontend should use default avatar image if not defined
+  avatar?: string; // Avatar uri (https link)
+  network: string; // Chain id
+  safeAddress: string; // Gnosis Safe address
+  votingToken: string; // Voting token address
 
   /**
-   * Get proposal by proposal id
-   * @param proposalId proposal id
+   * Get the list of zDAO assets and amount in USD
+   * @returns assets in zDAO
    */
-  getProposalById(proposalId: string): Promise<Proposal | undefined>;
+  listAssets(): Promise<zDAOAssets>;
 
   /**
-   * Get all the votes by proposal id filtering with the function parameter
-   * @param proposalId proposal id
-   * @param first voting count to fetch
-   * @param voter voter address to filter
-   * @param skip start index
+   * Get the list of zDAO transactions
+   * @returns list of transactions
    */
-  getProposalVotes(
-    proposalId: string,
-    { first, voter, skip }: any
-  ): Promise<Array<Vote>>;
+  listTransactions(): Promise<Transaction[]>;
 
   /**
-   * Get the result of proposal from votes
-   * @param zNA zNA address
-   * @param proposal proposal information
-   * @param votes list of votes to calculate result
+   * Get the list of the proposals created in the zDAO
+   * @param from start index
+   * @param count number of proposals to fetch
+   * @return list of proposals
    */
-  getProposalResults(
-    zNA: string,
-    proposal: any,
-    votes: any
-  ): Promise<{
-    resultsByVoteBalance: number;
-    sumOfResultsBalance: number;
-  }>;
+  listProposals(from: number, count: number): Promise<Proposal[]>;
 
   /**
-   * Get voting power of the user in zDAO
-   * @param zNA zNA address
-   * @param account account address
-   * @param proposal proposal information
-   * @returns voting power as number
+   * Get the specific proposal
+   * @param id proposal id
+   * @returns proposal instance
+   * @exception throw Error if not exist proposal id
    */
-  getVotingPower(zNA: string, account: string, proposal: any): Promise<number>;
+  getProposal(id: ProposalId): Promise<Proposal>;
 
   /**
    * Create a proposal in zDAO
-   * @param dao zDAO
+   * @param signer signer wallet
    * @param payload packaged parameters to create a proposal
    * @returns proposal id if success
    */
-  createProposal(dao: zDAO, payload: CreateProposalDto): Promise<string>;
+  createProposal(
+    signer: Wallet,
+    payload: CreateProposalParams
+  ): Promise<Proposal>;
+}
+
+export interface Proposal {
+  id: ProposalId; // proposal id
```

Comment:
Consider refactoring out all of the required "data" of this interface into a "properties" or "data" interface.
For instance you could do
```
export interface ProposalProperties {
  id: ProposalId; // proposal id
  type: string; // proposal type, by default 'single-choice'
  author: string; // proposal creator address
  title: string; // proposal title
  body?: string; // empty body if not defined
  ipfs: string; // uri to ipfs which contains proposal information and signature
  choices: VoteChoice[];
  created: Date;
  start: Date;
  end: Date;
  state: 'pending' | 'active' | 'closed';
  network: string; // chain id
  snapshot: string; // snapshot block number
  scores: number[]; // scores per all the choices
  votes: number; // number of voters
}

export interface Proposal extends ProposalProperties {
...
}
```
This also allows you to refactor the `Client` constructor:

```
constructor(
    zDAO: zDAOClient,
    snapshotClient: SnapshotClient,
    gnosisSafeClient: GnosisSafeClient,
    properties: ProposalProperties
  ) {
  ```

Which also will satisfy another comment I would add which is functions with more than ~4 parameters are unmaintainable (usually) and should use interface objects instead.